[docs] Update environment-variables.mdx (#23700)
expo/expo: docs/pages/guides/environment-variables.mdx

<Terminal cmd={['$ NODE_ENV=production npx expo start']} />

### Disabling environment variables

> Environment variables in Expo CLI have two parts, both can be disabled:

Reviewer: ```suggestion
Environment variables in Expo CLI have two parts and both can be disabled:
```

Using a `,` is not a separator. Let's break this down into two sentences or use conjunction like the above suggestion.

> 1. Expo CLI will load the `.env` files into the global process. This can be disabled by setting the environment variable `EXPO_NO_DOTENV` before any Expo CLI command is run, like `EXPO_NO_DOTENV=1`.

Reviewer: ```suggestion
1. Expo CLI automatically loads the **.env** files into the global process. To disable this behavior, set the environment variable `EXPO_NO_DOTENV` to `1` before running any Expo CLI command: `EXPO_NO_DOTENV=1`.
```

- [Filenames to be bold as per writing styleguide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#filenames-as-bold-text)
- [Using present tense](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#use-present-tense)
- Previously, it was suggested: "...by setting the environment variable `EXPO_NO_DOTENV`". This was missing as to what the env var needs to be set to or if just setting the env var is enough.

> 2. Expo's Metro config will inline environment variables in the client JavaScript bundle during serialization. This can be disabled with `EXPO_NO_CLIENT_ENV_VARS`, like `EXPO_NO_CLIENT_ENV_VARS=1`.

Reviewer: ```suggestion
2. Expo's Metro config includes the inline serialization of environment variables in the client JavaScript bundle. To disable this behavior, you can use `EXPO_NO_CLIENT_ENV_VARS=1`.
```

- `inline environment variables in the client` was confusing as to what does inline meant in that context. Tried to update the phrase as above.
- one thing missing from above is whether `EXPO_NO_CLIENT_ENV_VARS=1` has to be run before an Expo CLI command (like in point 1)?

> If you're experiencing issues with environment variables, you can try disabling one or both of these features. Consider contacting support on GitHub or Discord if you're still having issues.

Reviewer: ```suggestion
If you're experiencing issues with environment variables, you can try disabling one or both of these features.
```

There is no need to explicitly mention where to contact support. Every page in docs explicitly provide link to Discord, forums, etc.
